utils/oomparser/oomparser_test.go

```diff
@@ -0,0 +1,204 @@
+	currentOomInstance := new(OomInstance)
+	err := getContainerName(startLine, currentOomInstance)
+	if err != nil {
+		t.Errorf("bad line fed to getContainerName should yield no error, but had error %v", err)
+	}
+	if currentOomInstance.ContainerName != "" {
+		t.Errorf("bad line fed to getContainerName yielded no container name but set it to %s", currentOomInstance.ContainerName)
+	}
+	err = getContainerName(containerLine, currentOomInstance)
+	if err != nil {
+		t.Errorf("container line fed to getContainerName should yield no error, but had error %v", err)
+	}
+	if currentOomInstance.ContainerName != "/mem2" {
+		t.Errorf("getContainerName should have set containerName to /mem2, not %s", currentOomInstance.ContainerName)
+	}
+}
+
+func TestGetProcessNamePid(t *testing.T) {
+	currentOomInstance := new(OomInstance)
+	couldParseLine, err := getProcessNamePid(startLine, currentOomInstance)
+	if err != nil {
+		t.Errorf("bad line fed to getProcessNamePid should yield no error, but had error %v", err)
+	}
+	if couldParseLine {
+		t.Errorf("bad line fed to getProcessNamePid should return false but returned %v", couldParseLine)
+	}
+
+	correctTime, err := time.Parse(time.Stamp, "Jan 21 22:01:49")
+	couldParseLine, err = getProcessNamePid(endLine, currentOomInstance)
+	if err != nil {
+		t.Errorf("good line fed to getProcessNamePid should yield no error, but had error %v", err)
+	}
+	if !couldParseLine {
+		t.Errorf("good line fed to getProcessNamePid should return true but returned %v", couldParseLine)
+	}
+	if currentOomInstance.ProcessName != "evilprogram2" {
+		t.Errorf("getProcessNamePid should have set processName to evilprogram2, not %s", currentOomInstance.ProcessName)
+	}
+	if currentOomInstance.Pid != 19667 {
+		t.Errorf("getProcessNamePid should have set PID to 19667, not %d", currentOomInstance.Pid)
+	}
+	if !correctTime.Equal(currentOomInstance.TimeOfDeath) {
+		t.Errorf("getProcessNamePid should have set date to %v, not %v", correctTime, currentOomInstance.Pid)
+	}
+}
+
+func TestCheckIfStartOfMessages(t *testing.T) {
+	couldParseLine, err := checkIfStartOfOomMessages(endLine)
+	if err != nil {
+		t.Errorf("bad line fed to checkIfStartOfMessages should yield no error, but had error %v", err)
+	}
+	if couldParseLine {
+		t.Errorf("bad line fed to checkIfStartOfMessages should return false but returned %v", couldParseLine)
+	}
+
+	couldParseLine, err = checkIfStartOfOomMessages(startLine)
+	if err != nil {
+		t.Errorf("start line fed to checkIfStartOfMessages should yield no error, but had error %v", err)
+	}
+	if !couldParseLine {
+		t.Errorf("start line fed to checkIfStartOfMessages should return true but returned %v", couldParseLine)
+	}
+}
+
+func TestAnalyzeLines(t *testing.T) {
+	expectedContainerOomInstance, err := createExpectedContainerOomInstance()
+	if err != nil {
+		t.Errorf("got error when trying to create container oom instance: %v", err)
+	}
+	helpTestAnalyzeLines(expectedContainerOomInstance, containerLogFile, t)
```

Comment:
Lets separate this tests into two so it is easier to see what is failing:
- TestAnalyzeLinesSystemOom
- TestAnalyzeLinesContainerOom